lib/core/board.rb

```diff
@@ -25,6 +25,18 @@ def invalid_cell(position)
     !@grid[position].is_empty?
   end
 
+  def all_empty_cells
+    @grid.find_all { |cell| cell.is_empty? }
+  end
+
+  def available_positions
```

Comment:
Isn't this the opposite of `all_empty_cells`? In that case, you could implement it the same way in a one-liner.